hyperspy/drawing/markers.py

```diff
@@ -0,0 +1,676 @@
+            happen.
+        """
+        if signal_axes is None or (
+            signal_axes == "metadata"
+            and not signal.metadata.has_item("Peaks.signal_axes")
+        ):
+            new_signal = signal
+        elif signal_axes == "metadata" and signal.metadata.has_item(
+            "Peaks.signal_axes"
+        ):
+            new_signal = signal.map(
+                convert_positions,
+                inplace=False,
+                ragged=True,
+                signal_axes=signal.metadata.Peaks.signal_axes,
+            )
+        elif isinstance(signal_axes, (tuple, list)):
+            new_signal = signal.map(
+                convert_positions, inplace=False, ragged=True, signal_axes=signal_axes
+            )
+        else:
+            raise ValueError(
+                "The keyword argument `signal_axes` must be one of 'metadata' a"
+                "tuple of `DataAxes` or None"
+            )
+        kwargs[key] = new_signal.data
+        return cls(collection_class=collection_class, **kwargs)
+
+    def _get_data_shape(self):
+        for key, item in self.kwargs.items():
+            if is_iterating(item):
+                return item.shape
+        return ()
+
+    def __deepcopy__(self, memo):
+        new_marker = markers2collection(self._to_dictionary())
+        return new_marker
+
+    def _to_dictionary(self):
+        marker_dict = {
+            "marker_type": "MarkerCollection",
+            "collection_class": self.collection_class,
+            "plot_on_signal": self._plot_on_signal,
+            "kwargs": self.kwargs,
+        }
+        return marker_dict
+
+    def get_data_position(self, get_static_kwargs=True):
+        """
+        Return the current keyword arguments for updating the collection.
+        """
+        current_keys = {}
+        if self.is_iterating:
+            indices = self.axes_manager.indices[::-1]
+            for key, value in self.kwargs.items():
+                if is_iterating(value):
+                    if key not in self.dask_kwargs:
+                        val = value[indices]
+                        # some keys values need to iterate
+                        if key in ["sizes", "color"] and not hasattr(val, "__len__"):
+                            val = (val,)
+                        current_keys[key] = val
+                elif get_static_kwargs:
+                    current_keys[key] = value
+                else:  # key already set in init
+                    pass
+            if len(self.dask_kwargs) > 0:
+                current_keys.update(self._get_cache_dask_kwargs_chunk(indices))
+        else:
+            current_keys = self.kwargs
+        return current_keys
+
+    def update(self):
+        if not self.is_iterating:
+            return
+        else:
+            kwds = self.get_data_position(get_static_kwargs=False)
+            self.collection.set(**kwds)
+
+    def _initialize_collection(self):
```

Comment:
Hmm this is kind of possible.  It requires a couple of changes.

1. We would have to remove the scatter function in favor of using the Circles collection.  I like the scatter function because it gives lots of flexibility without the pain of setting things up yourself. That being said it probably does make sense.
2. The Quiver function would have to not take `ax` as an argument.  Honestly it shouldn't anyways and it should be moved to `matplotlib.collections`

So the answer is that is probably the right thing to do...